### PR TITLE
Add metrics reporters conf value check

### PR DESCRIPTION
**Pull request by zhenjiaguo**
### _Why are the changes needed?_

Configuration metrics reporter need to check that match kyuubi provided reporter.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/MetricsConf.scala [hunk 1]
```diff
@@ -46,6 +46,8 @@ object MetricsConf {
     .stringConf
     .transform(_.toUpperCase())
     .toSequence()
+    .checkValue(_.forall(ReporterType.values.map(_.toString).contains),
```
Comment (yaooqinn) [inline kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/MetricsConf.scala:49]: use checkValues？

Comment (zhenjiaguo) [reply kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/MetricsConf.scala:49]: `checkValues` is used one of pre-defined set, not a sequence, like EngineType, Either `SPARK_SQL` or `FLINK_SQL`. ReporterType is conf as sequence, like `CONSOLE,JMX`. So I don't use checkValue method.

By the way, I see a lot of sequence use `checkValue` method, maybe we should add new check method called `checkSeqValues`.

Comment (zhenjiaguo) [reply kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/MetricsConf.scala:49]: I found it hard to implement the method `checkSeqValues`. Leave it.

**Review by yaooqinn — APPROVED** [review]

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**